feat(datapoints): add support for data quality (beta)

23andreas opened the pull request:
Add support for data quality status codes to beta package

https://developer.cognite.com/dev/concepts/resource_types/timeseries#data-point-quality-status-codes

neringaalt committed:
feat: update alerts filter (#1070)

* feat: update alerts filter

* fix: use DateRange

cognite-cicd committed:
chore(release): publish new package versions [skip ci]

 - @cognite/sdk-beta@5.7.0

23andreas committed:
feat(beta-datapoints): add support for data quality indicators to retrieve method

23andreas commented on packages/beta/src/types.ts:
```diff
@@ -322,3 +330,72 @@ export interface Subscription {
+}
+
+export interface DatapointStatus {
+  code: number;
+  symbol: string;
```
Symbol can be a combination of [Category](https://developer.cognite.com/dev/concepts/reference/quality_codes/#subcategories)
and [modifier flags.](https://developer.cognite.com/dev/concepts/reference/quality_codes/#modifier-flags) 
For example `BadBrowseNameInvalid, StructureChanged, Low`. 

I initially tried typing it using [template literals](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html)
but this scenario was too advanced for that, so went with string. 

Code is the number you get by adding all the category and modifier codes together. 
The example above would result in 2153775104 + 32768 + (1024) + 256 = 2153809152. 
[More details here](https://developer.cognite.com/dev/concepts/reference/quality_codes/#status-code-structure)

sagar-thalwar committed:
feat(alpha): add sort option for filtering simulation routine revision (#1072)

* feat(alpha): add sort option for filtering simulation routine revision

* chore(ci): deploy using cd environment secret

cognite-cicd committed:
chore(release): publish new package versions [skip ci]

 - @cognite/sdk-alpha@0.21.0

polomani commented on packages/beta/src/types.ts:
```diff
@@ -322,3 +330,72 @@ export interface Subscription {
+
+export interface DatapointAggregate extends DatapointAggregateStable {
+  countUncertain?: number;
+  countBad?: number;
```
this is pretty clean! nice

BugGambit commented on packages/stable/src/__tests__/api/datamodels.int.spec.ts:
```diff
@@ -48,8 +48,17 @@ describe('Data models integration test', () => {
+      // Filter spaces last updated more than 24 hours ago
+      (space) => space.lastUpdatedTime < Date.now() - 24 * 60 * 60 * 1000
+    );
+    await client.spaces.delete(oldSpaces.map((space) => space.space));
```
Move to a function to avoid duplicate code? It's currently repeated three times

23andreas replied:
Good suggestion 
will do that

bkuzma committed:
feat: add simulator logs API (#1073)

cognite-cicd committed:
chore(release): publish new package versions [skip ci]

 - @cognite/sdk-alpha@0.22.0

bkuzma committed:
feat: add logId to SimulatorModelRevision (#1075)

* feat: add logId to SimulatorModelRevision

* feat: make logId required

cognite-cicd committed:
chore(release): publish new package versions [skip ci]

 - @cognite/sdk-alpha@0.23.0

23andreas committed:
refactor: rename dataPointsApi parent folder

23andreas committed:
refactor(dataPointsApi): move getMonthsBetweenDates to class function

23andreas committed:
feat(dataPointsApi): beta add support for statatus codes

23andreas committed:
fix(beta): explicitly export types

abdullah-cognite committed:
chore: update schema of SimulatorRoutineRevision (#1076)

* simint: support api breaking changes

* update

* update mock

bkuzma committed:
fix: fix simulator cron type (#1078)

* feat: fix readme typo

* fix: fix type

cognite-cicd committed:
chore(release): publish new package versions [skip ci]

 - @cognite/sdk-alpha@0.23.1

BugGambit committed:
chore: fix broken CI/CD pipeline (#1080)

* chore: fix broken CI/CD pipeline

* chore: rename 'COGNITE_AZURE_TENANT_ID' to 'COGNITE_AZURE_DOMAIN'

BugGambit committed:
feat: group membership (#1079)

* feat: group membership

* test: fix invalid request/response format

cognite-cicd committed:
chore(release): publish new package versions [skip ci]

 - @cognite/sdk-alpha@0.23.2
 - @cognite/sdk-beta@5.7.1
 - @cognite/sdk-playground@7.0.15
 - @cognite/sdk@9.11.0

bkuzma committed:
feat: add enabled property to data sampling (#1081)

* feat: add enabled property to data sampling

* fix: fix test

---------

Co-authored-by: cognite-bulldozer[bot] <51074376+cognite-bulldozer[bot]@users.noreply.github.com>

cognite-cicd committed:
chore(release): publish new package versions [skip ci]

 - @cognite/sdk-alpha@0.24.0

BugGambit committed:
test: clean up old FDM spaces (#1083)

* test: clean up old FDM spaces

* test: fix raise condition for DM

BugGambit committed:
fix(groups): inconsistent types compared to the API spec (#1082)

cognite-cicd committed:
chore(release): publish new package versions [skip ci]

 - @cognite/sdk-alpha@0.24.1
 - @cognite/sdk-beta@5.7.2
 - @cognite/sdk-playground@7.0.16
 - @cognite/sdk@9.11.1

BugGambit committed:
fix(Group): explicitly define sourceId for all group variants (#1084)

cognite-cicd committed:
chore(release): publish new package versions [skip ci]

 - @cognite/sdk-alpha@0.24.2
 - @cognite/sdk-beta@5.7.3
 - @cognite/sdk-playground@7.0.17
 - @cognite/sdk@9.11.2

danlevings committed:
feat: introduce remaining instance endpoints (#1085)

* feat: introduce remaining instance endpoints

* fix: update descriptions, test tests

* fix: snippets

* fix: remove test-permissions workflow

* fix: make delete endpoint work over limit of 1000

* fix: snippet errors and test errors

* fix: snippets

* fix: snippet

* fix: use new .delete in test utils

cognite-cicd committed:
chore(release): publish new package versions [skip ci]

 - @cognite/sdk-alpha@0.24.3
 - @cognite/sdk-beta@5.7.4
 - @cognite/sdk-playground@7.0.18
 - @cognite/sdk@9.12.0

23andreas committed:
refactor: fix type

23andreas committed:
fix: wrong api path

23andreas committed:
test: add data point test for status codes

23andreas commented on packages/beta/src/__tests__/api/dataPoints.int.spec.ts:
```diff
@@ -0,0 +1,611 @@
+    await client.timeseries.delete([{ id: timeserie.id }]);
+  });
+
+  test('retrieve aggregates with correct data point status counts', async () => {
```
Test #1 for status codes, rest of the tests are copied from stable

23andreas commented on packages/beta/src/__tests__/api/dataPoints.int.spec.ts:
```diff
@@ -0,0 +1,611 @@
+    expect(response[0].datapoints[4].countBad).toBe(0);
+  });
+
+  test('retrieve datapoints with status code', async () => {
```
test #2

polomani commented on packages/beta/src/__tests__/api/dataPoints.int.spec.ts:
```diff
@@ -0,0 +1,611 @@
+  const timestampFrom = Date.now() - 60 * 60 * 48 * 1000;
+  const timestampTill = Date.now() - 60 * 60 * 24 * 1000;
+  const datapoints = [
+    {
```
maybe we should not copy-paste all the existing tests? only add a new one with the statuses here?

polomani commented on packages/beta/src/api/dataPoints/dataPointsApi.ts:
````diff
@@ -0,0 +1,107 @@
+   * const monthlyAggregatesData = await client.datapoints.retrieveDatapointMonthlyAggregates({ items: [{ id: 123 }] });
+   * ```
+   */
+  public retrieveDatapointMonthlyAggregates = async (
````
instead of copying the implementation here, could you just make a 

```
protected doRetrieveDatapointMonthlyAggregates = async <T extends DatapointAggregates[] = DatapointAggregates[]> (
```

then inside
```
public retrieveDatapointMonthlyAggregates
```
you would just call
```
doRetrieveDatapointMonthlyAggregates(...)
```
same here in beta, you'd just call
```
doRetrieveDatapointMonthlyAggregates<DatapointAggregates[]>
```

with your new type as a param

polomani reviewed (approved):
LGTM

23andreas committed:
Merge branch 'master' into as/beta-time-series-data-quality

23andreas committed:
refactor: remove duplicate import
